test_competitor/src/test_competitor.cpp

```diff
@@ -164,6 +176,59 @@ void TestCompetitor::right_bins_camera_cb(
   right_bins_camera_pose_ = msg->sensor_pose;
 }
 
+void TestCompetitor::conveyor_parts_cb(
+    const ariac_msgs::msg::ConveyorParts::ConstSharedPtr msg
+)
+{
+  if (!conveyor_parts_recieved_data)
+  {
+    RCLCPP_INFO(get_logger(), "Received data from conveyor parts");
+    conveyor_parts_recieved_data = true;
+  }
+
+  conveyor_parts_expected_ = msg->parts;
+}
+
+void TestCompetitor::conveyor_camera_cb(
+    const ariac_msgs::msg::AdvancedLogicalCameraImage::ConstSharedPtr msg)
+{
+  if (!conveyor_camera_recieved_data)
+  {
+    RCLCPP_INFO(get_logger(), "Received data from conveyor camera");
+    conveyor_camera_recieved_data = true;
+  }
+  
+  conveyor_part_detected_ = msg->part_poses;
+  conveyor_camera_pose_ = msg->sensor_pose;
+}
+
+void TestCompetitor::breakbeam_cb(
+    const ariac_msgs::msg::BreakBeamStatus::ConstSharedPtr msg)
+{
+    if (!breakbeam_received_data)
+    {
+      RCLCPP_INFO(get_logger(), "Received data from conveyor breakbeam");
+      breakbeam_received_data = true;
+    }
+
+    breakbeam_status = msg->object_detected;
+
+    if (breakbeam_status)
+    {
+      // Lock conveyor_parts_ mutex
+      std::lock_guard<std::mutex> lock(conveyor_parts_mutex);
+
+      for (auto part : conveyor_part_detected_)
```

Comment:
This should only ever add one part to the list. Check that the y position of the pose is close to the breakbeam pose to make sure the correct part is being added.

Reply:
I'm currently executing this check within the 'if' condition block.

Reply:
Implemented in comit [4c8c8d6](https://github.com/usnistgov/ARIAC/pull/262/commits/4c8c8d6688d4db90e4d75ed890c9e16005a66497)